src/JestProcessManagement/JestProcess.ts

```diff
@@ -35,8 +35,10 @@ export class JestProcess {
         if (--this.keepAliveCounter > 0) {
           this.runner.removeAllListeners()
           this.startRunner()
-        } else if (this.onExitCallback) {
-          this.onExitCallback(this)
+        } else {
+          if (this.onExitCallback) {
+            this.onExitCallback(this)
+          }
           if (this.stopRequested) {
```

Comment:
now that we do set this.resolve to null, I think we should check `this.resolve` instead of `this.stopRequested` here. As long as there is a`this.resolve`, we should just resolve it.

Reply:
Okay, checking here for `this.resolve` which i rename to `this.stopResolveCallback` (maybe better naming?) and turning `this.stopRequested` into method.